R/package.R

```diff
@@ -117,6 +117,12 @@ do_check <- function(targz, package, args, libpath, repos,
   profile <- make_fake_profile(session_output = session_output)
   on.exit(unlink(profile), add = TRUE)
 
+  # if the pkg.Rcheck directory already exists, unlink it
+  check_dir <- paste0(package, ".Rcheck")
+  if (file.exists(check_dir)) {
```

Comment:
Sure that is fair, race conditions.

Reply:
Yeah, it might even make sense to `filelock::filelock()` the check dir, in case you call multiple checks from multiple sessions, and `unlock()` in on.exit. But we don't have to do that in this PR.